sql/src/main/java/org/apache/druid/sql/calcite/schema/MetadataSegmentView.java

```diff
@@ -73,8 +72,14 @@
   private final BrokerSegmentWatcherConfig segmentWatcherConfig;
 
   private final boolean isCacheEnabled;
+  private final Object lock = new Object();
+  /**
+   * Use {@link ImmutableSortedSet} so that the order of segments is deterministic and
+   * sys.segments queries return the segments in sorted order based on segmentId
+   */
   @Nullable
```

Comment:
Please annotate `@MonotonicNonNull` instead of `@Nullable`

Reply:
ok, good to know about this annotation.